core/extensions/extension.hpp

```diff
@@ -23,6 +24,8 @@ namespace kagome::extensions {
 
    public:
     virtual ~Extension() = default;
+
+    virtual std::shared_ptr<runtime::WasmMemory> memory() const = 0;
```

Comment:
if memory is required, maybe it is good to pass it by reference? why pointer?

Reply:
In most places memory is accepted by shared_ptr, so it would be easier to return shared_ptr here, and pass it as pointer rather than create a shared_ptr from the reference